### PR TITLE
feat: escape strings for debug output

### PR DESCRIPTION
**Pull request by mlutze**
related to #4664. Note that this does not work for strings inside of arrays or other objects

### CODE REVIEW

File: main/src/library/Prelude.flix [hunk 1]
```diff
@@ -131,6 +131,77 @@ pub def println(x: a): Unit \ IO with ToString[a] =
...
+                        discard append(sb, "\\u" + code);
+                        visit(i + 1)
+                    } else {
+                        discard append(sb, "${c}");
+                        visit(i + 1)
+                    }
+            }
+        }
+
+    };
```
Comment (magnus-madsen) [inline main/src/library/Prelude.flix:200]: I think this is overly complex. I would only focus on replacing \n and \r etc. and using some String.replaceAll.

The more fancy stuff we can add later and by need.

**Comment by JonathanStarup** [comment]
A fun implementation could also be to take the source program representation of the string

**Comment by magnus-madsen** [comment]
I would really just escape the escape chars.

**Comment by magnus-madsen** [comment]
I mean the opposite, replace newlines by escape chars etc.

**Pull request merged**